### ISSUE TITLE
goconst: hardcoded IgnoreTests: false is not equivalent to exclusion rules

### ISSUE THREAD

**Issue by franciscocpg** [golangci/golangci-lint#6478]
## Description

In [`pkg/golinters/goconst/goconst.go`](https://github.com/golangci/golangci-lint/blob/b3ac139dffb84c6e541422b0f73252e7a29313fc/pkg/golinters/goconst/goconst.go#L63-L64), `IgnoreTests` is hardcoded to `false` with the comment:

```go
// Should be managed with `linters.exclusions.rules`.
IgnoreTests: false,
```

The assumption is that users can use `linters.exclusions.rules` with `path: _test\.go` to achieve the same effect. **This is incorrect** — the two mechanisms are not equivalent, leading to non-deterministic (flaky) CI failures.

## Why they're not equivalent

1. With `IgnoreTests: false`, goconst counts occurrences across **all** files in the package (test + non-test)
2. goconst picks **one** file location to report the issue — the location depends on the order of `pass.Files` from `go/analysis`, which is non-deterministic across platforms (e.g., darwin/arm64 vs linux/amd64)
3. The exclusion rule `path: _test\.go` only checks whether that **single report location** matches

So if a string like `"email"` appears 2 times in `job.go` and 3 times in `job_test.go`:
- goconst finds 5 total occurrences (above the default threshold of 3) and reports **one** issue
- On **macOS**: goconst happens to report at `job_test.go` → exclusion rule matches → ✅ no error
- On **Linux CI**: goconst happens to report at `job.go` → exclusion rule doesn't match → ❌ flaky failure

With `IgnoreTests: true`, goconst would only count the 2 occurrences in `job.go`, which is below the threshold — no issue is generated at all.

## Reproduction

`.golangci.yml`:
```yaml
version: "2"
linters:
  enable:
    - goconst
  exclusions:
    rules:
      - linters:
          - goconst
        path: _test\.go
```

Given a package where a string literal appears a few times in production code (below threshold) and several times in test code (pushing total above threshold), the goconst issue will be reported or not depending on which file `pass.Files` lists first — which varies by OS/architecture.

## Suggested fix

Expose the `IgnoreTests` setting to users via `linters.settings.goconst.ignore-tests` (like other goconst settings are exposed), so that goconst can skip test files entirely during analysis rather than relying on post-hoc exclusion of the report location.

---
🤖 *Comment by Claude Code*

**Comment by franciscocpg** [comment]
Closing to follow the bug template
